src/Features/CSharp/Portable/NavigationBar/CSharpNavigationBarItemService.cs

```diff
@@ -182,18 +187,41 @@ private static bool IsAccessor(ISymbol member)
     }
 
     private static SymbolItem? CreateItemForMember(
-        Solution solution, ISymbol member, SyntaxTree tree, CancellationToken cancellationToken)
+        Solution solution, ISymbol member, SyntaxTree tree, SemanticModel semanticModel, CancellationToken cancellationToken)
     {
         var location = GetSymbolLocation(solution, member, tree, cancellationToken);
         if (location == null)
             return null;
 
+        using var _ = ArrayBuilder<RoslynNavigationBarItem>.GetInstance(out var localFunctionItems);
+        foreach (var syntaxReference in member.DeclaringSyntaxReferences)
+        {
+            if (syntaxReference.SyntaxTree != tree)
+            {
+                // The reference is not in this file, no need to include in the outline view.
+                continue;
+            }
+
+            var node = syntaxReference.GetSyntax(cancellationToken);
+            foreach (var localFunction in node.DescendantNodes().Where(CSharpSyntaxFacts.Instance.IsLocalFunctionStatement))
+            {
+                var localFunctionSymbol = semanticModel.GetDeclaredSymbol(localFunction, cancellationToken);
+                // Check to make sure we only include local functions that are directly contained in the current member.
```

Comment:
This feels n^2 how you have things now

Reply:
I think it possibly was.  I adjusted the implementation and should only check each local function once